src/main/java/fr/rader/boblite/Main.java

```diff
@@ -23,242 +29,123 @@ public void start() {
-                    }
+        if (tempFileDirectory == null) {
+            System.exit(0);
+        }
 
-                    continue;
-                }
+        // Okay, time to get working!
+
+        // Get the total number of system CPU threads.
+        int totalSystemThreadNumber = Runtime.getRuntime().availableProcessors();
+        int executorThreadNumber;
+        if (replays.size() < totalSystemThreadNumber){
+            executorThreadNumber = replays.size();
+        }else{
+            // Leave at least one system thread idle, don't make the computer unusable.
+            executorThreadNumber = totalSystemThreadNumber - 1;
+            // If for whatever reason the computer only has one CPU core/thread, just use one thread.
+            if(executorThreadNumber < 1) executorThreadNumber = 1;
+        }
 
-                // we write the packet timestamp, size, packet id
-                writer.writeInt(timestamp);
-                writer.writeInt(size);
-                writer.writeVarInt(packetID);
-                // and finally, we write the packet data
-                writer.writeByteArray(reader.readFollowingBytes(size - 1));
-            }
+        System.out.println("Using " + executorThreadNumber + " thread(s) for " + replays.size() + " task(s)...");
 
-            // we then flush the writer
-            writer.flush();
+        ExecutorService executor = Executors.newFixedThreadPool(totalSystemThreadNumber);
 
-            // changing the progress bar action text,
-            // as we're now doing something else
-            progressBar.setActionText("Writing the replay, this can take a while...");
+        long startTime = System.currentTimeMillis();
 
-            // we open the zip file, write the recording.tmcpr file and close the zip
-            replayZip.open();
-            replayZip.addFile(writer.getInputStream(), "recording.tmcpr");
-            replayZip.close();
+        // Submit all tasks.
+        for(ReplayData replay : replays){
+            executor.submit(new EditReplayTask(
+                    replay,
+                    tempFileDirectory,
+                    menu));
+        }
 
-            writer.clear();
+        // Wait for all task to be completed.
+        executor.shutdown();
+        boolean isDone = false;
+        while(!isDone) {
+            try {
+                // Timeout doesn't really matter in this use case as we will want to wait for all tasks to complete before terminating.
+                isDone = executor.awaitTermination(7, TimeUnit.DAYS);
+            } catch (InterruptedException ex) { }
+        }
 
-            // kill the progress bar window as it's now useless
-            progressBar.kill();
+        long finishTime = System.currentTimeMillis();
+        System.out.println("Competed " + replays.size() + " task(s) in " + (int)(((finishTime - startTime) / 1000) / 60) + "m " + (int)(((finishTime - startTime) / 1000) % 60) + "s");
```

Comment:
Typo in the first word

Reply:
Resolved in commit [fix typo - b0b80c5](https://github.com/RaderRMT/BobLite/pull/1/commits/b0b80c5263382ae977fcf69adf13c6bd63e469e7)